tests/Unit/API/Site/Controllers/Ads/RecommendationsControllerTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Site\Controllers\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AccountService;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads\RecommendationsController;
+use Automattic\WooCommerce\GoogleListingsAndAds\Exception\AccountReconnect;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\RESTControllerUnitTest;
+use Exception;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * Class RecommendationsControllerTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Site\Controllers\Ads
+ */
+class RecommendationsControllerTest extends RESTControllerUnitTest {
+
+	/** @var MockObject|AccountService $account */
+	protected $account;
+
+	/** @var RecommendationsController $controller */
+	protected $controller;
+
+	protected const ROUTE_RECOMMENDATIONS = '/wc/gla/ads/recommendations';
+
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->account    = $this->createMock( AccountService::class );
+		$this->controller = new RecommendationsController( $this->server, $this->account );
+		$this->controller->register();
+	}
+
+	public function test_get_recommendations_returns_all_recommendations() {
+		$this->account->method( 'get_connected_account' )
+			->willReturn( [ 'status' => 'connected' ] );
+
+		$response = $this->do_request( self::ROUTE_RECOMMENDATIONS, 'GET' );
+
+		$this->assertEquals( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+		$this->assertCount( 2, $data );
+		$this->assertEquals( 1, $data[0]['id'] );
+		$this->assertEquals( 2, $data[1]['id'] );
```

Comment:
Once you're no longer hard coding the response, you'll need to mock the `AdsRecommendations::get_recommendations` response as part of this test.